### PR TITLE
Display stickers on the rooms

### PR DESCRIPTION
**Pull request by Tyuoli**
vector-im/riot-meta#146
vector-im/riot-android#2097

### CODE REVIEW

**Comment by ylecollen** [comment]
see my comments in https://github.com/matrix-org/matrix-android-sdk/pull/247

File: vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java [hunk 1]
```diff
@@ -1263,20 +1275,33 @@ private View getImageVideoView(int type, final int position, View convertView, V
             MessageRow row = getItem(position);
             Event event = row.getEvent();
 
-            Message message;
+            Message message = new Message();
```
Comment (manuroe) [inline vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java]: The code was better before

File: vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java [hunk 2]
```diff
@@ -215,26 +214,34 @@ public VectorMessagesAdapter(MXSession session, Context context, MXMediasCache m
                 R.layout.adapter_item_vector_message_room_member,
                 R.layout.adapter_item_vector_message_text_emote_notice,
                 R.layout.adapter_item_vector_message_file,
-                R.layout.adapter_item_vector_message_image_video,
                 R.layout.adapter_item_vector_message_merge,
+                R.layout.adapter_item_vector_message_image_video,
```
Comment (manuroe) [inline vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java]: What did you have move `adapter_item_vector_message_image_video`?
It now appears twice.

File: vector/src/main/java/im/vector/adapters/VectorMessagesAdapterMediasHelper.java [hunk 3]
```diff
@@ -243,6 +245,37 @@ void managePendingImageVideoDownload(final View convertView, final Event event,
                     thumbHeight = videoMessage.info.thumbnail_info.h;
                 }
             }
+        } else if (message instanceof StickerMessage) {
```
Comment (manuroe) [inline vector/src/main/java/im/vector/adapters/VectorMessagesAdapterMediasHelper.java]: This code should be useless now, no?
Plus I am wondering if the condition at https://github.com/vector-im/riot-android/pull/2119/files#diff-195abc8f7f6b2882511ac16bee261c64R200 is true before going to here.

File: vector/src/main/java/im/vector/adapters/VectorMessagesAdapterMediasHelper.java [hunk 4]
```diff
@@ -265,8 +298,10 @@ void managePendingImageVideoDownload(final View convertView, final Event event,
         if (null == downloadId) {
             if (message instanceof VideoMessage) {
                 downloadId = mMediasCache.downloadIdFromUrl(((VideoMessage) message).getUrl());
-            } else {
+            } else if (message instanceof ImageMessage) {
                 downloadId = mMediasCache.downloadIdFromUrl(((ImageMessage) message).getUrl());
+            } else if (message instanceof StickerMessage) {
```
Comment (manuroe) [inline vector/src/main/java/im/vector/adapters/VectorMessagesAdapterMediasHelper.java]: same comment as above.

File: vector/src/main/java/im/vector/adapters/VectorMessagesAdapterMediasHelper.java [hunk 5]
```diff
@@ -424,13 +459,16 @@ void managePendingImageVideoUpload(final View convertView, final Event event, Me
...
-            if (isVideoMessage) {
+            if (message instanceof VideoMessage) {
                 uploadingUrl = ((VideoMessage) message).getUrl();
                 isUploadingContent = ((VideoMessage) message).isLocalContent();
 
-            } else {
+            } else if (message instanceof ImageMessage){
                 uploadingUrl = ((ImageMessage) message).getUrl();
                 isUploadingContent = ((ImageMessage) message).isLocalContent();
+            } else if (message instanceof StickerMessage) {
```
Comment (manuroe) [inline vector/src/main/java/im/vector/adapters/VectorMessagesAdapterMediasHelper.java]: same comment as above.

File: vector/src/main/java/im/vector/adapters/VectorMediasViewerAdapter.java [hunk 6]
```diff
@@ -174,7 +175,9 @@ private void downloadHighResMedia(final View view, final int position) {
                 imageInfo.mMimeType = "image/jpeg";
             }
             downloadHighResPict(view, position);
-        } else {
+
+        // video
+        } else  {
```
Comment (manuroe) [inline vector/src/main/java/im/vector/adapters/VectorMediasViewerAdapter.java]: extra space

**Pull request merged**